src/table/src/metadata.rs

```diff
@@ -358,6 +387,57 @@ impl TableMeta {
 
         Ok(meta_builder)
     }
+
+    fn split_requests_by_column_location<'a>(
+        &self,
+        table_name: &str,
+        requests: &'a [AddColumnRequest],
+    ) -> Result<SplitResult<'a>> {
+        let table_schema = &self.schema;
+        let mut columns_at_first = Vec::new();
+        let mut columns_location_map = HashMap::new();
+        let mut columns_at_last = Vec::with_capacity(requests.len());
```

Comment:
I think it's not needed to pre-allocated capacity here, since the columns to be added to the last is uncertain as well:

```suggestion
        let mut columns_at_last = Vec::new();
```